### PR TITLE
Add configuration panel for Application Credentials

### PR DESCRIPTION
**Pull request by allenporter**
## Proposed change

Add a configuration panel for Application Credentials which allow users to configure OAuth Client ID and OAuth Client
Secret for an integration using the user interface.

Work in progress:
- [x] determine correct place for this in the UI (arbitrary placeholder for now, asked for input in `#devs_us`)
- [x] add core support for an integration (e.g. xbox)
- [x] add core websocket for returning supported integrations
- [x] render integrations drop down or search in fronted UI
- [x] support for deleting an integration
- [ ] special display of imported credentials vs user added credentials

Capture

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion: https://github.com/home-assistant/core/pull/69148 https://github.com/home-assistant/architecture/discussions/692
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [X] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Comment by zsarnett** [comment]
@matthiasdebaat Let's talk about this. Not sure where this should be. The configuration sections are all changing so this will definitely not stay where is but maybe move to the system tab 🤷

**Comment by allenporter** [comment]
> @matthiasdebaat Let's talk about this. Not sure where this should be. The configuration sections are all changing so this will definitely not stay where is but maybe move to the system tab 🤷

Thanks, really appreciate this.  I'll continue to work on the functionality, thought just wanted to flag this early.

**Comment by allenporter** [comment]
> > @matthiasdebaat Let's talk about this. Not sure where this should be. The configuration sections are all changing so this will definitely not stay where is but maybe move to the system tab 🤷
> 
> Thanks, really appreciate this. I'll continue to work on the functionality, thought just wanted to flag this early.

@zsarnett i saw you landed a config revamp. Do you have a more well formed opinion now of where this should land, or otherwise have any principles regarding the most recent revamp?

I still have other pieces to iterate on fix up but thought I would ask while this was top of mind.

**Comment by allenporter** [comment]
> > > @matthiasdebaat Let's talk about this. Not sure where this should be. The configuration sections are all changing so this will definitely not stay where is but maybe move to the system tab 🤷
> > 
> > 
> > Thanks, really appreciate this. I'll continue to work on the functionality, thought just wanted to flag this early.
> 
> @zsarnett i saw you landed a config revamp. Do you have a more well formed opinion now of where this should land, or otherwise have any principles regarding the most recent revamp?
> 
> I still have other pieces to iterate on fix up but thought I would ask while this was top of mind.

I just rebased and can see how `System` makes sense given it seems like we've found a place to put low level things like this. I'm going to go ahead and move this there optimistically, given I think I can see where you're going with this, but happy to put this wherever makes sense.

**Comment by zsarnett** [comment]
Similar to the help icon except you will need to use an overflow button

https://github.com/home-assistant/frontend/blob/dev/src/panels/config/automation/ha-automation-picker.ts#L246-L251

Overflow Icon Button
https://github.com/home-assistant/frontend/blob/dev/src/panels/config/dashboard/ha-config-dashboard.ts#L157-L171

**Comment by zsarnett** [comment]
```html
  <ha-button-menu
    corner="BOTTOM_START"
    slot="toolbar-icon"
    activatable
  >
    <ha-icon-button
      slot="trigger"
      .label=${this.hass.localize("ui.common.menu")}
      .path=${mdiDotsVertical}
    ></ha-icon-button>
    <ha-clickable-list-item
      @click=${this._entryClicked}
      href="/config/lovelace/resources"
      aria-label=${this.hass.localize(
        "ui.panel.config.lovelace.resources.caption"
      )}
    >
      ${this.hass.localize(
        "ui.panel.config.lovelace.resources.caption"
      )}
    </ha-clickable-list-item>
  </ha-button-menu>
```

**Comment by allenporter** [comment]
Thank you that was extremely helpful. I am hitting a couple layout issues, still working through.

The overflow menu looks good ✅ 
Overflow menu

though it does not expand quite right ❓ 
Overflow menu expanded issue

And in narrow mode it looks good on 3 tabs ✅ 
Narrow overflow

However on the integrations tab it gets out of place ❌ 
Narrow overflow integrations issue

**Comment by zsarnett** [comment]
I'll try to take a look this weekend or Monday

**Comment by allenporter** [comment]
No rush, I also need to do one last selector for an integration drop down

**Comment by matthiasdebaat** [comment]
Awesome work @allenporter!

**Comment by zsarnett** [comment]
I messed up the git commands but should be all fixed up now. Your commits are all the first commit now... Sorry about that. But the overflow should be good

**Comment by allenporter** [comment]
@zsarnett 🔥 🔥🔥🔥!!!! thank you, so kind of you. I now will work on making the integration selector and take out of draft.

**Comment by zsarnett** [comment]
When the backend gets merged can you set it back to ready for review?

File: src/panels/config/application_credentials/dialog-add-application-credential.ts
Comment (zsarnett) [inline src/panels/config/application_credentials/dialog-add-application-credential.ts:34]: `mwc-list-item` Needs imported

File: src/panels/config/application_credentials/dialog-add-application-credential.ts
Comment (zsarnett) [inline src/panels/config/application_credentials/dialog-add-application-credential.ts:73]: can we do this in `showDialog()`

File: src/panels/config/application_credentials/dialog-add-application-credential.ts
Comment (zsarnett) [inline src/panels/config/application_credentials/dialog-add-application-credential.ts:118]: Use `ha-textfield`

File: src/panels/config/application_credentials/dialog-add-application-credential.ts
Comment (zsarnett) [inline src/panels/config/application_credentials/dialog-add-application-credential.ts:133]: Use `ha-textfield`

File: src/panels/config/application_credentials/dialog-add-application-credential.ts
Comment (zsarnett) [inline src/panels/config/application_credentials/dialog-add-application-credential.ts:169]: Set domains to undefined

File: src/panels/config/application_credentials/dialog-add-application-credential.ts
Comment (zsarnett) [inline src/panels/config/application_credentials/dialog-add-application-credential.ts:17]: Unused

File: src/panels/config/application_credentials/ha-config-application-credentials.ts
Comment (zsarnett) [inline src/panels/config/application_credentials/ha-config-application-credentials.ts:130]: use `ha-help-tooltip` or import `paper-tooltip`

File: src/panels/config/entities/entity-registry-settings.ts
Comment (zsarnett) [inline src/panels/config/entities/entity-registry-settings.ts:279]: hmm this got carried over it seems. Can you add this back.

File: src/panels/config/integrations/ha-integration-overflow-menu.ts
Comment (zsarnett) [inline src/panels/config/integrations/ha-integration-overflow-menu.ts:42]:
```suggestion
```

File: src/panels/config/application_credentials/dialog-add-application-credential.ts
Comment (allenporter) [reply src/panels/config/application_credentials/dialog-add-application-credential.ts]: Removed.

File: src/panels/config/application_credentials/dialog-add-application-credential.ts
Comment (allenporter) [reply src/panels/config/application_credentials/dialog-add-application-credential.ts:33]: Imported.

File: src/panels/config/application_credentials/dialog-add-application-credential.ts
Comment (allenporter) [reply src/panels/config/application_credentials/dialog-add-application-credential.ts]: Fixed.

File: src/panels/config/application_credentials/dialog-add-application-credential.ts
Comment (allenporter) [reply src/panels/config/application_credentials/dialog-add-application-credential.ts]: Fixed.

File: src/panels/config/application_credentials/dialog-add-application-credential.ts
Comment (allenporter) [reply src/panels/config/application_credentials/dialog-add-application-credential.ts:164]: Fixed.

File: src/panels/config/entities/entity-registry-settings.ts
Comment (allenporter) [reply src/panels/config/entities/entity-registry-settings.ts]: Whoops!

File: src/panels/config/application_credentials/dialog-add-application-credential.ts
Comment (allenporter) [reply src/panels/config/application_credentials/dialog-add-application-credential.ts]: Done.

File: src/panels/config/application_credentials/ha-config-application-credentials.ts
Comment (allenporter) [reply src/panels/config/application_credentials/ha-config-application-credentials.ts]: Done

**Comment by allenporter** [comment]
> When the backend gets merged can you set it back to ready for review?

Core PR said: `Great! Can be merged when frontend PR is happy with this.`
https://github.com/home-assistant/core/pull/69148#pullrequestreview-955848132

So figured i'd get your pulse before actually merging.

File: src/panels/config/application_credentials/dialog-add-application-credential.ts
Comment (bramkragten) [inline src/panels/config/application_credentials/dialog-add-application-credential.ts:164]: This should be a public function and fire `dialog-closed` event:

```ts
  public closeDialog() {
    this._params = undefined;
    this._domains = undefined;
    fireEvent(this, "dialog-closed", { dialog: this.localName });
  }
```

Comment (allenporter) [reply src/panels/config/application_credentials/dialog-add-application-credential.ts]: Fixed.

File: src/panels/config/application_credentials/dialog-add-application-credential.ts
Comment (bramkragten) [inline src/panels/config/application_credentials/dialog-add-application-credential.ts:69]: I think this is handled by the dialog, and this is not needed?

**Review by bramkragten — DISMISSED** [review]

File: src/panels/config/application_credentials/dialog-add-application-credential.ts
Comment (allenporter) [reply src/panels/config/application_credentials/dialog-add-application-credential.ts]: Removed.

**Comment by allenporter** [comment]
Whoops, noticed I had a lint error, fixed.

**Review by bramkragten — APPROVED** [review]

**Pull request merged**
